modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/IgniteCacheDatabaseSharedManager.java

```diff
@@ -622,28 +621,37 @@ private void validateConfiguration(DataStorageConfiguration memCfg) throws Ignit
     }
 
     /**
-     * Check wal archive size configuration for correctness.
+     * Check WAL archive size configuration for correctness.
      *
      * @param memCfg durable memory configuration for an Apache Ignite node.
+     * @throws IgniteCheckedException if maximum WAL archive size is configured incorrectly
      */
     private void checkWalArchiveSizeConfiguration(DataStorageConfiguration memCfg) throws IgniteCheckedException {
-        if (memCfg.getWalHistorySize() == DFLT_WAL_HISTORY_SIZE || memCfg.getWalHistorySize() == Integer.MAX_VALUE)
-            LT.warn(log, "DataRegionConfiguration.maxWalArchiveSize instead DataRegionConfiguration.walHistorySize " +
-            "would be used for removing old archive wal files");
-        else if (memCfg.getMaxWalArchiveSize() == DFLT_WAL_ARCHIVE_MAX_SIZE)
-            LT.warn(log, "walHistorySize was deprecated and does not have any effect anymore. " +
-                "maxWalArchiveSize should be used instead");
-        else
-            throw new IgniteCheckedException("Should be used only one of wal history size or max wal archive size." +
-                "(use DataRegionConfiguration.maxWalArchiveSize because DataRegionConfiguration.walHistorySize was deprecated)"
-            );
+        long maxWalArchiveSize = memCfg.getMaxWalArchiveSize();
 
-        if (memCfg.getMaxWalArchiveSize() != DataStorageConfiguration.UNLIMITED_WAL_ARCHIVE
-                && memCfg.getMaxWalArchiveSize() < memCfg.getWalSegmentSize())
-            throw new IgniteCheckedException(
-                "DataRegionConfiguration.maxWalArchiveSize should be greater than DataRegionConfiguration.walSegmentSize " +
-                    "or must be equal to " + DataStorageConfiguration.UNLIMITED_WAL_ARCHIVE + "."
+        if (!CU.isPersistenceEnabled(memCfg)) {
+            if (maxWalArchiveSize != DataStorageConfiguration.DFLT_WAL_ARCHIVE_MAX_SIZE) {
+                LT.info(log, "Maximum WAL archive size has been configured but this node has been launched in " +
+                    "non-persistent mode, so this parameter will be ignored");
+            }
+            return;
+        }
+
+        if (memCfg.isWalHistorySizeParameterUsed())
+            LT.warn(log,
+                "DataRegionConfiguration.walHistorySize property is deprecated and is no longer supported. " +
+                    "It will be ignored and DataRegionConfiguration.maxWalArchiveSize property will be used " +
+                    "instead to control eviction of archived WAL files"
```

Comment:
Lets use work 'removing' or similar here instead of eviction. Eviction term is already used in our configuration with different meanings.

Otherwise patch looks good to me.